content/hello-world-program.md

```diff
@@ -0,0 +1,328 @@
+
+### Add the instruction to the transaction
+
+Once we’ve built our instruction, we add the instruction to our transaction. This process can be repeated if a transaction requires multiple instructions.
+
+```tsx
+transaction.add(instruction)
+```
+
+### Submit the transaction
+
+Lastly, we send the new transaction to be processed by the Solana network.
+
+```tsx
+const transactionSignature = await web3.sendAndConfirmTransaction(
+    connection,
+    transaction,
+    [payer]
+)
+```
+
+...and there you have it - you now know all the things you need for the foundations of creating a Solana program using Rust. Let’s practice what we’ve learned so far!
+
+# Demo
+
+We're going to build a "Hello, World!" program using Solana Playground. Solana Playground is a tool that allows you to write and deploy Solana programs from the browser. Click [here](https://beta.solpg.io/) to open Solana Playground. Next, go ahead and delete everything in the default `lib.rs` file and create a Playground wallet.
+
+![Gif Solana Playground Create Wallet](../assets/hello-world-create-wallet.gif)
+
+First, let's bring into scope everything we’ll need from the `solana_program` crate.
+
+```rust
+use solana_program::{
+    account_info::AccountInfo,
+    entrypoint,
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+    msg
+};
+```
+
+Next, let's set up the entry point to our program using the `entrypoint!` macro and create the `process_instruction` function. The `msg!` macro then allows us to print “Hello, world!” to the program log when the program is invoked.
+
+```rust
+entrypoint!(process_instruction);
+
+pub fn process_instruction(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8]
+) -> ProgramResult{
+    msg!("Hello, world!");
+
+    Ok(())
+}
+```
+
+All together, the “Hello, world!” program will look like this:
+
+```rust
+use solana_program::{
+    account_info::AccountInfo,
+    entrypoint,
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+    msg
+};
+
+entrypoint!(process_instruction);
+
+pub fn process_instruction(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8]
+) -> ProgramResult{
+    msg!("Hello, world!");
+
+    Ok(())
+}
+```
+
+Now let's build and deploy our program using Solana Playground.
+
+![Gif Solana Playground Build and Deploy](../assets/hello-world-build-deploy.gif)
+
+Finally, let's invoke our program from the client side. Download the code [here](https://github.com/ZYJLiu/solana-hello-world-client).
+
+The focus of this lesson is to build our Solana program, so we’ve gone ahead and provided the client code to invoke our  “Hello, world!” program. The code provided includes a `sayHello` helper function that builds and submits our transaction. We then call `sayHello` in the main function and print a Solana Explorer URL to view our transaction details in the browser.
+
+In the `index.ts` file you should see the following:
```

Comment:
We don't need to put all of the code here I don't think. Just have them download the client, explain what it does, and have the run it.

Reply:
removed index.ts code block